libvast/src/system/source.cpp

```diff
@@ -139,15 +139,31 @@ void source_state::filter_and_push(
       VAST_DEBUG("{} forwards {}/{} produced {} events after filtering",
                  reader->name(), filtered_slice->rows(), unfiltered_rows,
                  slice.schema());
-      push_to_out(std::move(*filtered_slice));
+      auto import_ok = executor.add(std::move(*filtered_slice));
+      if (import_ok) {
+        VAST_WARN("source can not add slice to pipeline executor - data "
+                  "loss");
+      }
+      auto transformed_slices = executor.finish();
+      for (auto&& slice : *transformed_slices) {
```

Comment:
I think we should also check if the expected is set.
If not - print a warning and return.

Reply:
I applied the same `transformed_slices` checks to the `source` in https://github.com/tenzir/vast/pull/2896/commits/75e61108bb2d5a492bcb87bc207a2b603874d1b9